### PR TITLE
Homework 4

### PR DESCRIPTION
**Pull request by YevheniiLohatskyi**

### CODE REVIEW

File: src/com/homework/Operation.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    MUL('*', (a, b) -> a * b),
+    DIV('/', (a, b) -> a / b),
+    ROOT('_', (a, b) -> Math.pow(a, 1 / b)),
+    POW('^', Math::pow),
+    MOD_POW('~', (a, b) -> Math.pow((a + b) / a + 117, b));
+
+    private final char symbol;
+    private final DoubleBinaryOperator operator;
+
+    Operation(char symbol, DoubleBinaryOperator operator) {
```
Comment (Taniann) [inline src/com/homework/Operation.java]: add acces modificator. now you can use it  within that package

File: src/com/homework/Operation.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        this.symbol = symbol;
+        this.operator = operator;
+    }
+
+    public double getResult(double a, double b) {
+        return this.operator.applyAsDouble(a, b);
```
Comment (Taniann) [inline src/com/homework/Operation.java]: it is unnecessary to write "this" in that case

File: src/com/homework/Operation.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    public char getSymbol() {
+        return this.symbol;
```
Comment (Taniann) [inline src/com/homework/Operation.java]: same as above about "this"

Comment (YevheniiLohatskyi) [reply src/com/homework/Operation.java]: Please, consider changes. I cannot currently tag you in Google Sheets due to technical reasons

**Pull request closed without merging**